core/src/types.ts

```diff
@@ -391,6 +391,10 @@ export interface ITransactionParameters {
   to: string
 }
 
+export interface ITransactionParametersWithNonce extends ITransactionParameters {
+  nonce: number
+}
+
```

Comment:
rather than writing new interface, how about adding optional property into `ITransactionParameters`?
```
export interface ITransactionParameters {
  payload: string
  gasLimit: number | string
  to: string
  none?: number
}
```

Reply:
hmmm... interesting suggestion but I think making nonce optional might complicate things. If nonce is optional then we need to check if nonce is properly passed and if not then throw errors and handle it elsewhere. But if nonce if required we don't have to worry about human error (not passing the nonce while using that interface). Plus, I thought having a different interface emphasizes the existence of a manual nonce manipulation feature.

Reply:
well, I think I was thinking differently, if nonce is optional, functions utilizing nonce will check if nonce exist. if not it'll work same as before, if exists, it'll use passed nonce value. how do you think?

Reply:
As nonce is also part of transaction evm wise, I felt that it was a bit unnatural to separate types with nonce